### DISCUSSION TITLE
Non-ideal HWP rotation

### DISCUSSION THREAD

**Discussion by keskitalo** [simonsobs/sotodlib#312] [General]
## Simulating non-ideal HWP rotation

TOAST-3 now includes [an operator to perturb the HWP rotation](https://github.com/hpc4cmb/toast/blob/toast3/src/toast/ops/sim_hwp.py#L128).

It will be available to all SO workflows using the TOAST simulation facilities.

This discussion thread is about how to make the best use of the operator and how to improve it.

**Comment by keskitalo** [comment]
The current version of the operator supports two error modes:
* constant drift in scan rate over the observation
* Gaussian error in the encoder time stamps (AKA jitter)

@yukisakurai provided this lab-measured HWP spin rate PSD:
![image](https://user-images.githubusercontent.com/596250/196523924-28a116bc-8f82-44bf-b922-f5d7cacb1fc3.png)
Focus only on the blue PSD. It the difference in the encoder signal and stable 2Hz rotation.

**Reply by keskitalo** [reply]
@yukisakurai @msilvafe What are those spikes in the blue PSD? They certainly don't seem compatible with the two supported error modes.

**Reply by yukisakurai** [reply]
Thank you for your comment. We confirmed those spikes are harmonics, and they are almost same feature as Fig. 20 of the [PB2b paper](https://arxiv.org/pdf/2009.03972.pdf).
